src/util/simplify_expr_int.cpp

```diff
@@ -1478,19 +1372,147 @@ bool simplify_exprt::simplify_inequality(exprt &expr)
 
     expr.op0().swap(expr.op1());
 
-    // one is constant
-    simplify_inequality_constant(expr);
+    // RHS is constant, LHS is not
+    simplify_inequality_rhs_is_constant(expr);
     return false;
   }
   else if(tmp1_const)
   {
-    // one is constant
-    return simplify_inequality_constant(expr);
+    // RHS is constant, LHS is not
+    return simplify_inequality_rhs_is_constant(expr);
   }
   else
   {
     // both are not constant
-    return simplify_inequality_not_constant(expr);
+    return simplify_inequality_no_constant(expr);
+  }
+}
+
+/// simplifies inequalities for the case in which both sides
+/// of the inequality are constants
+bool simplify_exprt::simplify_inequality_both_constant(exprt &expr)
+{
+  PRECONDITION(expr.operands().size() == 2);
+
+  exprt tmp0 = expr.op0();
+  exprt tmp1 = expr.op1();
+
+  if(tmp0.type().id() == ID_c_enum_tag)
+    tmp0.type() = ns.follow_tag(to_c_enum_tag_type(tmp0.type()));
+
+  if(tmp1.type().id() == ID_c_enum_tag)
+    tmp1.type() = ns.follow_tag(to_c_enum_tag_type(tmp1.type()));
+
+  const auto &tmp0_const = to_constant_expr(tmp0);
+  const auto &tmp1_const = to_constant_expr(tmp1);
+
+  // are _both_ constant?
```

Comment:
Now this is moved, comment no longer needed

Reply:
done